app/forms/schools/amend_participant_cohort_form.rb

```diff
@@ -0,0 +1,116 @@
+              }
+    validates :target_cohort_start_year,
+              numericality: {
+                only_integer: true,
+                greater_than_or_equal_to: ECF_FIRST_YEAR,
+                less_than_or_equal_to: Date.current.year,
+                message: I18n.t("errors.cohort.invalid_start_year", start: ECF_FIRST_YEAR, end: Date.current.year),
+              },
+              exclusion: {
+                within: ->(form) { [form.source_cohort_start_year] },
+                message: ->(form, _) { I18n.t("errors.cohort.excluded_start_year", year: form.source_cohort_start_year) },
+              }
+    validates :target_cohort,
+              presence: {
+                message: lambda do |form, _|
+                  I18n.t("errors.cohort.blank", year: form.target_cohort_start_year, where: "the service")
+                end,
+              }
+    validates :participant_profile, presence: { message: I18n.t("errors.participant_profile.blank") }
+    validates :participant_declarations, absence: { message: I18n.t("errors.participant_declarations.exist") }
+    validates :induction_record,
+              presence: {
+                message: lambda do |form, _|
+                  I18n.t("errors.induction_record.blank", year: form.source_cohort_start_year)
+                end,
+              }
+    validates :target_school_cohort,
+              presence: {
+                message: lambda do |form, _|
+                  I18n.t("errors.cohort.blank", year: form.target_cohort_start_year, where: form.school&.name)
+                end,
+              }
+    delegate :school, to: :induction_record, allow_nil: true
+
+    def save
+      valid? && persist
+    end
+
+  private
+
+    def induction_programme
+      @induction_programme ||= target_school_cohort.default_induction_programme
+    end
+
+    def induction_record
+      return unless participant_profile
+
+      @induction_record ||= participant_profile.induction_records
+                                               .active_induction_status
+                                               .training_status_active
+                                               .joins(induction_programme: { school_cohort: :cohort })
+                                               .where(cohorts: { start_year: source_cohort_start_year })
+                                               .latest
+    end
+
+    def participant_declarations
+      @participant_declarations ||= participant_profile&.participant_declarations&.exists?
+    end
+
+    def participant_identity
+      @participant_identity ||= ParticipantIdentity.find_by_email(email)
+    end
+
+    def participant_profile
+      return unless participant_identity
+
+      @participant_profile ||= ParticipantProfile::ECF.training_status_active
+                                                      .active_record
+                                                      .find_by(participant_identity:)
+    end
+
+    def persist
```

Comment:
this method persist changes in the DB while the service (below) uses the form to run the business logic.
Should the service run the logic and the form use the service instead?
It seems weird that a service is using a form.
Is the form used in any page?

Reply:
I know what you mean. I wouldn't mind to change it but this is the rationale I built it this way:
To me Support people (via the Admin UI) should have an option on any participant to change it to a different cohort as they have to change a participant's name or email, etc. That is, individually at participant level. 
If we added it, that controller should consume this Induction::AmendParticipantCohortForm directly.

The service however, is something meant to replicate what we where doing manually at Zendesk (change a bunch of emails to a different cohort) and therefore it is consuming the form once per email.

Reply:
I get your point @ltello . The form will be needed in the admin and that's ok.
I think we should avoid that a service depends on a form.
I think it should be the opposite though.
The services we have all run business login called by controllers or forms.
Is there any issue in making the form call the service for the persist part?

Reply:
Form renamed to Service